realm/src/main/java/io/realm/internal/TableView.java

```diff
@@ -73,19 +73,24 @@
      * @param parent A table.
      * @param nativePtr pointer to table.
      */
-    protected TableView(Context context, Table parent, long nativePtr){
+    protected TableView(Context context, Table parent, long nativePtr) {
         this.context = context;
         this.parent = parent;
         this.nativePtr = nativePtr;
     }
 
+    protected TableView(Context context, Table parent, long nativePtr, TableQuery query) {
```

Comment:
I think the clean way to do this would be to have an `Table originatingTable` field in `TableQuery`, which would allow us to get rid of `parent` in this constructor